src/transformers/generation/utils.py

```diff
@@ -4310,8 +4314,14 @@ def assisted_decoding(
         ["It might be possible to get a better understanding of the nature of the problem, but it's not"]
         ```"""
         # Assistant: initialize assistant-related variables
-        if not hasattr(assistant_model, "max_assistant_tokens"):
-            assistant_model.max_assistant_tokens = 5  # this value, which will be updated, persists across calls
+        if hasattr(assistant_model, "max_assistant_tokens"):
+            warnings.warn(
+                "Setting `max_assistant_tokens` via `assistant_model.max_assistant_tokens` is deprecated and will be removed in v5. Make sure to set `max_assistant_tokens` via the generation_config instead.",
```

Comment:
Perhaps we can deprecate this earlier (like in v4.37)?

I haven't seen users fiddling with this internal variable :)

Reply:
Ok for me!